### ISSUE TITLE
Loss function of the monet

### ISSUE THREAD

**Issue by Guiliang** [baudm/MONet-pytorch#10]
In the file monet_model.py 
I believe line 123 should be
self.loss_mask = self.criterionKL(self.m_tilde_logits.**softmax**(dim=1), self.m)
instead of 
self.loss_mask = self.criterionKL(self.m_tilde_logits.**log_softmax**(dim=1), self.m)
since self.m is in the probability space and m_tilde should also be in the probability space not log-prob right?

**Comment by Guiliang** [comment]
I just realize the implementation of KLDivLoss in PyTorch is y_n (log y_n - x_n). It is abnormal. Maybe you are right, we should add a log to x_n.

**Comment by baudm** [comment]
> I just realize the implementation of KLDivLoss in PyTorch is y_n (log y_n - x_n). It is abnormal. Maybe you are right, we should add a log to x_n.

Yes, `torch.nn.KLDivLoss` expects the input to contain log-probabilities.
